pages/site-details.tsx

```diff
@@ -0,0 +1,69 @@
+import Details from '~/components/form/Details';
+import Location from '~/components/form/Location';
+import { useRouter } from 'next/router';
+
+import { FC, useState } from 'react';
+import { withSites } from '../lib/utils';
+
+enum Page {
+  Details = 'Details',
+  Location = 'Location',
+}
+
+interface Props {
+  page: Page;
+  setPage: (page: Page) => void;
+}
+
+const SiteDetailsDesktop: FC<Props> = ({ page, setPage }) => {
```

Comment:
This is the same as the function below named `SiteDetailsMobile`. I think you did too good of a job abstracting the logic, maybe remove one of these and rename the remaining one to `SiteForm` and use it in both occurances of the old ones.